includes/admin/class-wc-rest-payments-terminal-locations-controller.php

```diff
@@ -45,17 +45,41 @@ public function register_routes() {
 	 * @return WP_REST_Response|WP_Error
 	 */
 	public function get_store_location( $request ) {
-		try {
-			$name    = get_bloginfo();
-			$address = [
-				'city'        => WC()->countries->get_base_city(),
-				'country'     => WC()->countries->get_base_country(),
-				'line1'       => WC()->countries->get_base_address(),
-				'line2'       => WC()->countries->get_base_address_2(),
-				'postal_code' => WC()->countries->get_base_postcode(),
-				'state'       => WC()->countries->get_base_state(),
-			];
+		$store_address    = WC()->countries;
+		$location_address = array_filter(
+			[
+				'city'        => $store_address->get_base_city(),
+				'country'     => $store_address->get_base_country(),
+				'line1'       => $store_address->get_base_address(),
+				'line2'       => $store_address->get_base_address_2(),
+				'postal_code' => $store_address->get_base_postcode(),
+				'state'       => $store_address->get_base_state(),
+			]
+		);
 
+		// If address is not populated, emit an error and specify the URL where this can be done.
+		$is_address_unpopulated = 2 === count( $location_address ) && isset( $location_address['country'], $location_address['state'] );
+		if ( $is_address_unpopulated ) {
+			return rest_ensure_response(
+				new \WP_Error(
+					'store_address_is_incomplete',
+					__( 'The store address is incomplete, please actualize settings.', 'woocommerce-payments' ),
```

Comment:
May I suggest a different text here? 'The store address is incomplete, please update your settings.'

Reply:
Done in 8771a2109afd38c00e14d069927117af4498e376